### PR TITLE
Add telemetry

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
Add telemetry to Zoom.

The telemetry tracked is:
- Every time a meeting is created
- Every time a meeting is not created because there is another recent meeting
- Every time a meeting creation is forced (even if there is a recent meeting)
- Every time a user sign in into Zoom
- Every time a user signs out of Zoom
- Every time the plugin settings change whether it uses JWT or OAuth

We use telemetry to help improve the quality of Mattermost and to make design decisions for future releases.

This PR will be reviewed by the product team who will confirm which telemetry will actually be used to make those design decisions.

#### Ticket Link
None

### CODE REVIEW

**Comment by aaronrothschild** [comment]
Can we record HOW a meeting is started (ie: via Button vs Slash command)?

**Comment by larkox** [comment]
@aaronrothschild Yes. We can and we already do :wink:

**Comment by aaronrothschild** [comment]
Perfect, seems like the key actions are tracked.

**Review by mickmister — APPROVED** [review]
LGTM :+1:

File: server/telemetry.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+func (p *Plugin) trackOAuthModeChange(method string) {
+	p.tracker.TrackEvent("oauth_mode_change", map[string]interface{}{
+		"method": method,
+	})
+}
+
+func (p *Plugin) trackMeetingStart(userID, source string) {
+	p.tracker.TrackUserEvent("start_meeting", userID, map[string]interface{}{
```
Comment (mickmister) [inline server/telemetry.go:26]: Looks like there is a lint error for not checking the returned error from these calls

**Review by iomodo — APPROVED** [review]
Looks great @larkox !

**Comment by mickmister** [comment]
@larkox Looks like there are a couple of remaining lint errors

```
server/command.go:131:10: wrapperFunc: use strings.ReplaceAll method in `strings.Replace(helpText, "|", "`", -1)` (gocritic)
	text += strings.Replace(helpText, "|", "`", -1)
	        ^
server/command.go:134:18: wrapperFunc: use strings.ReplaceAll method in `strings.Replace(oAuthHelpText, "|", "`", -1)` (gocritic)
		text += "\n" + strings.Replace(oAuthHelpText, "|", "`", -1)
```

**Comment by DHaussermann** [comment]
@aaronrothschild Testing for this is ongoing now and looking good. 
One question for you... It seems that we return data for either `OAUTH` or `JWT` But, when the method comes back as oAuth we're not capturing data on the new Beta feature for app level OAuth.

Is it worth delaying this PR now to see if @larkox  can add this for us or should I create a new issue to see if this is something we'd like to add in the future?

**Comment by DHaussermann** [comment]
@larkox I have spoken to Aaron and he would like this change to also include `OAuth by Account Level App` in the data.
Is it possible to return this in Rudder?

**Comment by larkox** [comment]
@DHaussermann Telemetry added for using account level app. Whenever the setting changes (even if JWT is selected) a telemetry message will be sent. Nevertheless, since we are removing JWT soon, there will be no need to make more fine grain distinction.

**Comment by DHaussermann** [comment]
@larkox I may be missing something here but I'm not sure how I can tell if account level is enabled. 
- Deployed this Zoom build 
- Toggle account level oAuth True and False several times
- Waited for Rudder to show the data
**Observed** The `zoom_auth_menthod_change` event does not show me if account level is enabled or not
Screen Shot 2020-12-02 at 9 02 00 AM

**Comment by DHaussermann** [comment]
@larkox please let me know if you've had a chance to to review this.

**Comment by DHaussermann** [comment]
Ignore last comment. There was an oversight on my part. the data for account level does get captured.

**Review by DHaussermann — APPROVED** [review]
Tested and passed
- Confirmed all telemetry events are working as expected. 
- Functionality was added to differentiate between account level and user level oAuth

LGTM!

**Pull request merged**